compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4077,6 +4079,8 @@ private STNode parseTerminalExpression(SyntaxKind kind, STNode annots, boolean i
                 // In the checking action, nested actions are allowed. And that's the only
                 // place where actions are allowed within an action or an expression.
                 return parseCheckExpression(isRhsExpr, allowActions, isInConditionalExpr);
+            case FAIL_KEYWORD:
+                return parseFailExpression(isRhsExpr, allowActions, isInConditionalExpr);
```

Comment:
Since we add a new expression, we need to update few other places such as:
- [isExpression(...)](https://github.com/ballerina-platform/ballerina-lang/blob/b661950974d341878719e39adc19f7fbc1378b4b/compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java#L13424)
- [isValidExpressionStart(...)](https://github.com/ballerina-platform/ballerina-lang/blob/b661950974d341878719e39adc19f7fbc1378b4b/compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java#L10628)
- [isValidExprStart(...)](https://github.com/ballerina-platform/ballerina-lang/blob/b661950974d341878719e39adc19f7fbc1378b4b/compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java#L4213)
etc..

Reply:
Done.